### PR TITLE
Fix paypal adaptive transactions fx rate

### PR DESCRIPTION
**Pull request by Betree**
Fix https://github.com/opencollective/opencollective/issues/4021

All cases left in DB where PayPal expenses **paid manually**.

I believe I remember a bug where we would use `USD` as the default currency somewhere in the "Mark as paid" process, which probably messed up most of these expenses. I was not able to find the relevant issue/PR.

There are a few (mostly, `opencollective` inc) where I can't really explain what happens, the FX rate was wrong even though we were only using `USD`.

This PR:
- Fixes the abovementioned issue by considering that inputted amounts were using the source currency. We're therefore just re-computing them with FX rate set to `1`.
- Prevents paying PayPal expenses with adaptive when `host.currency` !== `collective.currency`. We know this can cause problems, and since we're https://github.com/opencollective/opencollective/issues/5923 it doesn't seem worth working on a fix.
- Adds a `transaction.data.isManual` flag on manually paid expenses, as it would have simplified the investigation to have an easy flag to surface that these expenses were paid manually

# Changelog

When run, the script will impact the following:

```
Transaction group b6f982d2-7f6b-41a1-80bf-0472e116fa31 (host=kendraio-org, collective=kendraio)
Transaction group d0c995fd-9c80-4a79-984c-7b1f9459e031 (host=flarum-foundation, collective=flarum)
Transaction group 12fc34fc-5345-485b-810f-6ea9c4eac7f5 (host=flarum-foundation, collective=flarum)
Transaction group ffd45053-1fa9-4f6b-b75b-a53585124d6c (host=europe, collective=sunbeam-city)
Transaction group ea33817d-a1ac-4138-af46-8979194bf22b (host=allforclimate, collective=xr-belgium)
Transaction group 8aa7e13a-6dba-4333-9a8c-bd11e0cc535c (host=europe, collective=sunbeam-city)
Transaction group 0c562334-3ec4-47b0-87e1-a1c358f0dab5 (host=europe, collective=tealwiki)
Transaction group e7102964-fc96-4b1c-9ff9-7caa6583f3e5 (host=opencollective, collective=opencollective)
Transaction group 48cb4241-1015-4737-bf87-df57e8caba46 (host=opencollective, collective=opencollective)
Transaction group cde123ed-b898-4c6f-9b78-73da7742b035 (host=opencollective, collective=opencollective)
Transaction group 79623c31-fdc8-4c79-a097-51149297a3d0 (host=opencollective, collective=opencollective)
Transaction group b788acf2-23e4-4cad-b97e-58d63383da4a (host=opencollective, collective=opencollective)
Transaction group d998e12b-a1de-421c-8f79-9cc55802d393 (host=opencollective, collective=opencollective)
Transaction group 064d5388-c3ce-4bf2-b88a-432fd49a6948 (host=opencollective, collective=opencollective)
Transaction group 50152e15-e30d-436c-9b86-36c69a5c7a82 (host=opencollective, collective=opencollective)
Transaction group a23787c7-bff8-4c86-aba7-d21d84d2301b (host=opencollective, collective=opencollective)
Transaction group b0fe6963-86a1-44e9-8563-1bdc1fa6d7cc (host=opencollective, collective=opencollective)
Transaction group 342e32ce-016a-4e10-ade2-b84acfdb2bf9 (host=opencollective, collective=opencollective)
Transaction group 6823b948-83bc-4245-ba74-857e04d96fee (host=opencollective, collective=opencollective)
Transaction group 8ec96c9c-35bd-460c-b1cb-60e54ae43d27 (host=opencollective, collective=opencollective)
Transaction group 63da9eb1-973b-4e26-a6a8-f119e4763dc4 (host=opencollective, collective=opencollective)
Transaction group 9afdfd4c-f1f7-4e71-b693-23825a2588bc (host=opencollective, collective=opencollective)
Transaction group 61729d7d-e73f-434d-9827-4cd0123cb7e2 (host=opencollective, collective=opencollective)
Transaction group bc7a6510-f40f-44b8-b640-4081b3e1b92e (host=opencollective, collective=opencollective)
Transaction group fee6ed49-f706-446b-98f9-be598dc42d43 (host=opencollective, collective=opencollective)
Transaction group fe4b1348-9311-4bea-8281-6bd7b690e7b9 (host=opencollective, collective=opencollective)
Transaction group 3e1a9a05-8d20-4103-bed4-d2a4791ccab4 (host=opencollective, collective=opencollective)
Transaction group 4a1d7667-5b7c-4d6f-a524-323cddadaf1d (host=opencollective, collective=opencollective)
Transaction group ac1c8682-3f33-4cfa-8d4e-7d5e1018e0ec (host=opencollective, collective=opencollective)
Transaction group 911088ce-72f7-4c0e-ab00-a8239469673f (host=opencollective, collective=opencollective)
Transaction group 19808cbc-3494-4890-9975-c340f4505421 (host=opencollective, collective=opencollective)
Transaction group b438e294-f4f3-4030-bcb1-9429d122cfde (host=opencollective, collective=opencollective)
Transaction group b919e3a2-4df7-45c7-bf86-4c415c47c8b8 (host=opencollective, collective=opencollective)
Transaction group 2512a8cc-a4f4-4dc9-a2ae-5efa63bfd090 (host=europe, collective=huneeds)
```

### CODE REVIEW

File: scripts/paypal/fix-adaptive-fx-rate.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    // Some of them were not set, `validate` does not like that
+    debit.hostFeeInHostCurrency = debit.hostFeeInHostCurrency || 0;
+    debit.platformFeeInHostCurrency = debit.platformFeeInHostCurrency || 0;
+    credit.hostFeeInHostCurrency = credit.hostFeeInHostCurrency || 0;
+    credit.platformFeeInHostCurrency = credit.platformFeeInHostCurrency || 0;
+
+    // Validate
+    try {
+      await models.Transaction.validate(debit, { validateOppositeTransaction: false });
+      await models.Transaction.validate(credit, { validateOppositeTransaction: false });
```
Comment (znarf) [inline scripts/paypal/fix-adaptive-fx-rate.ts:91]: Nice

File: server/graphql/common/expenses.ts [hunk 2]
```diff
@@ -1876,6 +1876,12 @@ export async function payExpense(req: express.Request, args: Record<string, unkn
     try {
       // Pay expense based on chosen payout method
       if (payoutMethodType === PayoutMethodTypes.PAYPAL) {
+        if (expense.collective.currency !== host.currency) {
```
Comment (znarf) [inline server/graphql/common/expenses.ts:1879]: Ok!

**Review by znarf — APPROVED** [review]
Looks good to me!

**Comment by Betree** [comment]
Script ran successfully, deploying the fix to prod

**Pull request merged**